shell/browser/electron_browser_context.cc

```diff
@@ -189,6 +194,7 @@ void ElectronBrowserContext::InitPrefs() {
   registry->RegisterFilePathPref(prefs::kDownloadDefaultDirectory,
                                  download_dir);
   registry->RegisterDictionaryPref(prefs::kDevToolsFileSystemPaths);
+  registry->RegisterDictionaryPref(kHidGrantedDevicesPref);
```

Comment:
Is this persistent..?

Reply:
yes.  The chromium implementation works the same way.

Reply:
Hmm, generally I prefer that Electron avoids persisting data across restarts. Would it be reasonable to expose this to app code instead, & then the app can make that determination?

Reply:
@nornagon I added this logic because folks are expecting our web implementations to work like Chromium eg someone logged this issue for Web Serial: https://github.com/electron/electron/issues/29366, which I fixed in https://github.com/electron/electron/pull/30209

Reply:
@jkleinsc hm, I feel the same way about #30209 as I feel about this, assuming these are both persistent across restarts.

The approach of saving them to disk works well for Chrome because there is user-accessible UI to inspect and modify those lists. Electron has no such UI, nor at present any APIs on which an app could build such a UI, so I think persisting like this is a little problematic. I think we should either:

1. expose these functions directly to the app (i.e. whenever we need a list of what serial/hid devices are allowed, or to know whether a particular device is allowed, ask the app), and not store anything to disk ourselves, or
2. add APIs that allow an app to access and modify the persisted preferences.

I think we should probably do one of these even if these preferences are only in-memory (i.e. not persisted across restarts), but I don't feel as strongly about it if that's the case. Remembering access to a particular device for the duration of a session feels OK.